frontend/src/views/WheresMyGene/components/GeneSearchBar/components/ShareButton/index.tsx

```diff
@@ -0,0 +1,97 @@
+import { Intent } from "@blueprintjs/core";
+import { IconNames } from "@blueprintjs/icons";
+import { useCallback, useContext, useEffect } from "react";
+import { track } from "src/common/analytics";
+import { EVENTS } from "src/common/analytics/events";
+import { usePrimaryFilterDimensions } from "src/common/queries/wheresMyGene";
+import { isSSR } from "src/common/utils/isSSR";
+import Toast from "src/views/Collection/components/Toast";
+import {
+  DispatchContext,
+  StateContext,
+} from "src/views/WheresMyGene/common/store";
+import { StyledButtonIcon } from "../QuickSelect/style";
+import { EXCLUDE_IN_SCREENSHOT_CLASS_NAME } from "../SaveImage";
+import { StyledButtonDiv, StyledLabel } from "./style";
+import { generateAndCopyShareUrl, loadStateFromQueryParams } from "./utils";
+
+export default function ShareButton(): JSX.Element {
+  const state = useContext(StateContext);
+  const { selectedFilters, selectedTissues, selectedGenes } = state;
+  const { isLoading: isLoadingFilterDims } = usePrimaryFilterDimensions();
+  const dispatch = useContext(DispatchContext);
+  // const [showURLCopyNotification, setShowURLCopyNotification] = useState(0);
+
+  const copyShareUrl = useCallback(() => {
+    if (!dispatch) return;
+    generateAndCopyShareUrl(selectedFilters, selectedTissues, selectedGenes);
+    track(EVENTS.WMG_SHARE_CLICKED, {
+      tissues: selectedTissues,
+      genes: selectedGenes,
+      dataset_filter: selectedFilters.datasets,
+      disease_filter: selectedFilters.diseases,
+      self_reported_ethnicity_filter: selectedFilters.ethnicities,
+      sex_filter: selectedFilters.sexes,
+      group_by_option: null, //TODO: add group by filter when work is completed
+    });
+    // setShowURLCopyNotification((prev) => prev + 1);
+    Toast.show({
+      icon: IconNames.LINK,
+      intent: Intent.PRIMARY,
+      message: "Share link copied",
+      timeout: 1000,
+    });
+  }, [selectedFilters, selectedTissues, selectedGenes, dispatch]);
+
+  useEffect(() => {
+    if (isSSR() || isLoadingFilterDims || !dispatch) return;
```

Comment:
Just asking for my knowledge, why did we need these isSSR checks?

Reply:
Window will return undefined if the page is being rendered on the server.